src/views/Task/Update.js

```diff
@@ -1,29 +1,192 @@
+        return false
+      }
+    }
+    return true
+  }
+
+
+  /**
+* Renders input fields based on definitions in fields[]
+* @param {String} label
+* @param {Mixed} type
+* @param {Method} validator
+* @param {boolean} isTextArea
+* @return {JSX} an input surrounded with a label
+*/
+  fieldRender([label, type, validator, isTextArea]) {
+    const name = camelcase(label);
+    if (isTextArea === true) {
+      return (
+        <label key={name}>
+          {label}
+          <textarea
+            name={name}
+            value={this.state[name]}
+            onChange={this.handleChange}
+            className={validator(this.state[name]) ? "valid" : ""}
+            required
+          />
+        </label>
+
+      )
+    }
+
+    return (
+      <label key={name}>
+        {label}
+        <input
+          name={name}
+          type={type}
+          value={this.state[name]}
+          onChange={this.handleChange}
+          className={validator(this.state[name]) ? "valid" : ""}
+          required
+        />
+      </label>
+    )
+  }
+
+
+  /**
+     * Sets state data when changes are made in text-inputs
+     * @param {Event} event
+     */
+  handleChange = event => {
+    this.setState({
+      [event.target.name]: event.target.value
+    })
+
+  }
+
+  componentDidMount = () => {
+    this.getTask()
+  }
+
+  /**
+   * Gets Task arguments and Setstate
+   */
+  getTask = async () => {
+    const task = await task.get(this.props.match.params.id);
+    this.setState({
+      title: task.title,
+      description: task.description,
+      urgency: task.urgency,
+      startDate: task.startDate,
+      endDate: task.endDate,
+    })
+  }
+
+  /**
+   * Sets the selected  date
+   * @param {Date} date
+   */
+  updateStartDate = (date) => {
+    this.setState({
+      startDate: date,
+      endDate: date,
+    })
+
+  }
+
+  /**
+   * Sets selected date of deadline
+   * @param {Date} date
+   */
+  updateEndDate = (date) => {
+    this.setState({ endDate: date })
+
+  }
+
+  /**
+   * Sets state data when changes are made in text-inputs
+   * @param {Event} event
+   */
+  handleChange = event => {
+    this.setState({
+      [event.target.name]: event.target.value
+    })
+
+  }
+
+  render() {
+    return (
+      <Layout>
+        <section>
+          <h2>Task</h2>
+          {this.state.error && (
+            <Alert>{this.state.error} </Alert>
+          )}
+          <form onUpdate={this.updateHandler}>
+            {this.fields.map(this.fieldRender.bind(this))}
+            <label>
+              Starting Date
+            <DatePicker onChange={this.updateStartDate} minDate={moment()} selected={this.state.startDate} />
+            </label>
+            <label>
+              Deadline
+            <DatePicker onChange={this.updateEndDate} minDate={this.state.startDate} selected={this.state.endDate} />
+            </label>
+          </form>
+          <input type="Update" value="Update" onClick={this.UpdateHandler} />
```

Comment:
input type should be = "submit". 
onClick has a casing-error, should be this.updateHandler, so it matches the method 😄

Reply:
updateHandler functionality works but returns an "unexpected end of json input" error for some reason 🤔